airflow/hooks/bash.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import signal
+from subprocess import PIPE, STDOUT, Popen
+from tempfile import TemporaryDirectory, gettempdir
+from typing import Dict, Optional
+
+from airflow import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class BashHook(BaseHook):
```

Comment:
So two observationsj:

This hook has nothing to do with bash -- it doesn't run bash, it just runs a command.

Secondly the main purpose of a Hook is to make use of a connection -- this doesn't do that either.

So I like the idea of separation, but I'm not sure of the exact separation/code layout.

One monent.

Reply:
Oh almost nothing to do with bash :)

Reply:
I don't know where this should live, but I think the class name should be something like `CommandRunner`.

I guess we could call this a Hook even if we don't use any connections -- the only other option I can think of is to have this under airflow.utils.

Reply:
hook > utils in my opinion. Maybe in future we will add some connections for auth (for example for gcp or aws)

Reply:
operators == orchestrator of task
hooks == vocabulary of operators

i had this thought two as i typed out a hook with no init params.  but i don't think that connections are really what defines a hook.  it's the collection of methods used for interacting with a resource.  there could be many hooks that don't need to worry about auth.

operators do one thing.  hooks can potentially do very many things.  so the things that have general use should get pushed down to a hook that can be reused.

Reply:
that said, i am not at all opposed to making this a CommandRunner -- that's is all it's doing after all.

just saying, i don't think that absence of connections is decisive

Reply:
@dstandish I agree, I always look at hooks as repository of methods reused by hooks. The layer of authentication is just a part of it but not crucial in my opinion.

Reply:
Let's keep this as a hook - runner is already a concept in airflow (check out StandardTaskRunner) so we don't want to overload that name